### PR TITLE
MB-11592 make ntsr shipment weight conditional

### PR DESCRIPTION
**Pull request by michaelvaldes-truss**
## Summary

This pr will make the shipment weight on the office shipment form required for TOO users when editing the shipment details.
When testing the changes to the storybook story, please make use of the storybook "controls" feature to test accessing the page as different user roles

## Setup to Run Your Code

<details>
<summary>💻 You will need to use two separate terminals to test this locally.</summary>

##### Terminal 1

Start the office UI locally.

```sh
make office_client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Log in to the office app as a TOO and view a move with an NTSR shipment (e.g. move code "NTSSUB" in devseed sub-scenario NTS_AND_NTSR)
2. Edit the NTSR shipment in the move. Ensure that the shipment weight field is required (displays "required" and restricts saving the form when empty).

## Verification Steps for Author

These are to be checked by the author.

- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### CODE REVIEW

**Review by YanZ777 — COMMENTED** [review]
I think the storybook stories for ShipmentWeightInput need to be updated as well if you're adding a required prop to ShipmentWeightInput.

**Review by rogeruiz — APPROVED** [review]
lgtm 🌈

Good work on this @michaelvaldes-truss this works locally for me as expected. I left a comment, but choose to update it in this PR or another as you want.

File: src/components/Customer/MtoShipmentForm/getShipmentOptions.js [hunk 1]
```diff
@@ -51,6 +51,7 @@ const ntsReleaseShipmentCounselorSchema = Yup.object().shape({
 
 const ntsReleaseShipmentTOOSchema = Yup.object().shape({
   delivery: RequiredPlaceSchema,
+  ntsRecordedWeight: Yup.string().required('Required'),
```
[the remark this answers is not in the document]
Comment (michaelvaldes-truss) [reply src/components/Customer/MtoShipmentForm/getShipmentOptions.js:54]: @rogeruiz I pushed a storybook update just as you left your comment :)

**Review by YanZ777 — APPROVED** [review]

**Review by amanda5iu — APPROVED** [review]
Works as expected!

**Pull request merged**